### PR TITLE
IGNITE-14054 : Improve discovery ducktest: add partial network drop.

### PR DESCRIPTION
**Pull request by Vladsz83**
Discovery ducktape test lacks simulation of partial network failure. Also it doesn't check disabled recoveryTimeout (connRecoveryTimeout==0). Raised by tickets:

[Infinite node persistance in the ring while outcoming connections are lost](https://issues.apache.org/jira/browse/IGNITE-14068)
[Remove duplicated ping: processing and raising StatusCheckMessage](https://issues.apache.org/jira/browse/IGNITE-13980)

### CODE REVIEW

File: modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py [hunk 1]
```diff
@@ -318,15 +333,17 @@ def drop_network(self, nodes=None):
         dsc_ports = str(dsc_spi.port) if not hasattr(dsc_spi, 'port_range') or dsc_spi.port_range < 1 else str(
             dsc_spi.port) + ':' + str(dsc_spi.port + dsc_spi.port_range)
 
-        cmd = f"sudo iptables -I %s 1 -p tcp -m multiport --dport {dsc_ports},{cm_ports} -j DROP"
+        if net_part in (IgniteAwareService.NetPart.ALL, IgniteAwareService.NetPart.INCOMING):
+            node.account.ssh_client.exec_command(
+                f"sudo iptables -I INPUT 1 -p tcp -m multiport --dport {dsc_ports},{cm_ports} -j DROP")
+            node.account.ssh_client.exec_command(
+                f"sudo iptables -I FORWARD 1 -p tcp -m multiport --dport {dsc_ports},{cm_ports} -j DROP")
```
Comment (anton-vinogradov) [inline modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py:340]: Why forward is incoming while it's outcoming as well?

File: modules/ducktests/tests/ignitetest/tests/discovery_test.py [hunk 2]
```diff
@@ -169,6 +177,9 @@ def _perform_node_fail_scenario(self, test_config):
             if LATEST_2_7 < test_config.version <= V_2_9_0:
                 discovery_spi.so_linger = 0
 
+            if test_config.disable_conn_recovery:
+                discovery_spi.connRecoveryTimeout = 0
```
Comment (anton-vinogradov) [inline modules/ducktests/tests/ignitetest/tests/discovery_test.py:181]: any guarantee it's not null before? :)
should we use explicit connRecoveryTimeout instead?

File: modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py [hunk 3]
```diff
@@ -33,13 +34,22 @@
...
     """
     The base class to build services aware of Ignite.
     """
+    @constructible
+    class NetPart(IntEnum):
+        """
+        Network part to emulate failure.
+        """
+        INCOMING = 0
+        OUTCOMING = 1
```
[the remark this answers is not in the document]
Comment (Vladsz83) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py]: Fixed

File: modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py [hunk 4]
```diff
@@ -33,13 +34,22 @@
...
+        ALL = 2
```
[the remark this answers is not in the document]
Comment (Vladsz83) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py:52]: There is not 'ALL'. And we are not interrested in separated FORWARD.

File: modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py (hunk 1 above)
Comment (Vladsz83) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py]: Fixed

File: modules/ducktests/tests/ignitetest/tests/discovery_test.py (hunk 2 above)
Comment (Vladsz83) [reply modules/ducktests/tests/ignitetest/tests/discovery_test.py:181]: null is checked in the template. pulint doesn't like many params. Gotta be disabled every time. Think it is ok for simple config.

**Pull request merged**